riscv-target-parser/src/lib.rs

```diff
@@ -135,11 +143,15 @@ impl RiscvTarget {
             })
         {
             if let Some(feature) = target_feature.strip_prefix('+') {
-                let extension = Extension::try_from(feature)?;
-                target.extensions.insert(extension);
+                if !not_an_isa_extension(feature) {
```

Comment:
To avoid double negative clauses, I would suggest you to change the function to `is_isa_extension` instead of `not_an_isa_extension`:

```suggestion
                if is_isa_extension(feature) {
```

Reply:
The idea with this naming was that the check in the new helper function is not exhaustive, `!not_an_isa_extension(f)` doesn't really imply that `f` is an extension. New feature flags like "relax" could be added.

That said, I agree that this is hard to read. I'm applying the change you suggest and will extend the code comment about `is_isa_extension` to mention this.